### PR TITLE
Explicitly check if object exists before creating new one

### LINKED ISSUE #76: Primary Key restraints issue with `object` table
This whole bug sounds doesn't sound like it should be possible, but I'm seeing this on a server I was testing on. Prior to commit 1211ed1344b073341910aa14cceab517a2209e97 the object table contained one namespace/object-id combination. This is dictated by the primary key restraint placed on the table. However, after updating to this commit we now have a `status` column. The problem I've somehow hit in my testing is that I get "impossible to happen" rows in mysql like:
~~~
   namespace    object_id    status
   default      deadbeef     UPLOADING
   default      deadbeef     UPLOADED
~~~
This is violating primary key restraints, but somehow mysql allowed the new row to be created. After that mysql goes nuts and can't do a select statement like: `select * from object where namespace = "default" and object_id = "deadbeef";`. It actually returns 0 rows.

If I take a DB dump and try to import it into a clean mariadb container, mysql will refuse to allow it.

I'm not sure the exact intent of this commit. However, I think one of 2 things needs to happen:
1) ensure you *update* status of an existing row and keep with the one row per object approach
2) add "status" as a primary key restraint to the table and that might lead to some other findQuery type changes I haven't thought through.

### PR DESCRIPTION
**Pull request by simao**
https://jira.mariadb.org/browse/MDEV-14837

Fixes #76

### CODE REVIEW

File: src/main/scala/com/advancedtelematic/treehub/db/ObjectRepository.scala [hunk 1]
```diff
@@ -41,7 +51,7 @@ protected class ObjectRepository()(implicit db: Database, ec: ExecutionContext)
   def update(namespace: Namespace, id: ObjectId, size: Long, status: ObjectStatus): Future[Unit] = {
     val io = findQuery(namespace, id)
       .map { r => r.size -> r.status }
-      .update((size, ObjectStatus.UPLOADED))
+      .update((size, status))
```
Comment (jochenschneider) [inline src/main/scala/com/advancedtelematic/treehub/db/ObjectRepository.scala]: nice

Comment (simao) [reply src/main/scala/com/advancedtelematic/treehub/db/ObjectRepository.scala]: ja. lucky we were only calling this method with that value!

Comment (jochenschneider) [reply src/main/scala/com/advancedtelematic/treehub/db/ObjectRepository.scala]: ;-)

**Comment by doanac** [comment]
thanks guys!

**Comment by doanac** [comment]
You guys ready to merge this yet? I'd like to get that fixed and then rebase/retest https://github.com/advancedtelematic/treehub/pull/75

**Pull request merged**